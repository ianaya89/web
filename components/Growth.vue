<template>
  <div :class="mainClasses">
    <span :class="[growthClasses, iconClass]" />
    <span v-if="showValue" class="ml-1 pt-1" :class="[applyColorOnText ? textColorClass : '']">
      <slot>
        <span ref="bounty" class="flex" />
      </slot>
    </span>
  </div>
</template>

<script>
import bounty from 'bounty'
import { bountyFilterFix } from '@/services/helpers'

export default {
  name: 'Growth',

  props: {
    mainClasses: {
      type: String,
      default: 'flex items-center'
    },
    upIcon: {
      type: String,
      default: 'fas fa-arrow-alt-circle-up'
    },
    downIcon: {
      type: String,
      default: 'fas fa-arrow-alt-circle-down'
    },
    iconClass: {
      type: String,
      default: ''
    },
    upColor: {
      type: String,
      default: 'text-green'
    },
    downColor: {
      type: String,
      default: 'text-red'
    },
    applyColorOnText: {
      type: Boolean,
      default: true
    },
    showValue: {
      type: Boolean,
      default: true
    },
    value: {
      type: Number,
      required: true
    },
    duration: {
      type: Number,
      required: false,
      default: 2000
    }
  },

  data: () => ({
    animated: false
  }),

  computed: {
    growthClasses() {
      if (this.value < 0) {
        return this.downIcon + ' ' + this.textColorClass
      } else {
        return this.upIcon + ' ' + this.textColorClass
      }
    },

    textColorClass() {
      if (this.value < 0) {
        return this.downColor
      } else {
        return this.upColor
      }
    }
  },

  watch: {
    value() {
      if (this.$refs.bounty != null) {
        const bountyInstance = bounty({
          el: this.$refs.bounty,
          value: this.$filters.percentage(this.value),
          initialValue: this.$filters.percentage(this.value),
          duration: 2000
        })
        if (!this.animated) {
          this.animated = !this.animated
          setTimeout(() => {
            bountyFilterFix(this.$refs.bounty)
          }, 2000)
        } else {
          if (bountyInstance) {
            bountyInstance.cancel()
          }
          bountyFilterFix(this.$refs.bounty)
        }
      }
    }
  }
}
</script>

<style lang="postcss">
span > svg {
  @apply .my-auto;
}
.text-red svg text {
  fill: #ff553f;
}
.text-green svg text {
  fill: #21ce99;
}
</style>
