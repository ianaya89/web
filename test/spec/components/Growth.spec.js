import { shallowMount } from '@vue/test-utils'
import Growth from '@/components/Growth'

describe('Growth', () => {
  it('renders Growth with value: 30', () => {
    const wrapper = shallowMount(Growth, {
      propsData: {
        value: 30
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders Growth with value: -30', () => {
    const wrapper = shallowMount(Growth, {
      propsData: {
        value: -30
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders Growth not colored text', () => {
    const wrapper = shallowMount(Growth, {
      propsData: {
        value: -30,
        applyColorOnText: false
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
