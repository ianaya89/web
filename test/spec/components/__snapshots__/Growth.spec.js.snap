// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Growth renders Growth not colored text 1`] = `
<div
  class="flex items-center"
>
  <span
    class="fas fa-arrow-alt-circle-down text-red"
  />
   
  <span
    class="ml-1 pt-1"
  >
    <span
      class="flex"
    />
  </span>
</div>
`;

exports[`Growth renders Growth with value: -30 1`] = `
<div
  class="flex items-center"
>
  <span
    class="fas fa-arrow-alt-circle-down text-red"
  />
   
  <span
    class="ml-1 pt-1 text-red"
  >
    <span
      class="flex"
    />
  </span>
</div>
`;

exports[`Growth renders Growth with value: 30 1`] = `
<div
  class="flex items-center"
>
  <span
    class="fas fa-arrow-alt-circle-up text-green"
  />
   
  <span
    class="ml-1 pt-1 text-green"
  >
    <span
      class="flex"
    />
  </span>
</div>
`;
